src/signals/IncidentMap/components/IncidentLayer/IncidentLayer.tsx

```diff
@@ -73,15 +71,24 @@ export const IncidentLayer = ({
         })
       },
       pointToLayer: (incident: Incident, latlng) => {
-        if (selectedMarkerRef.current) {
-          resetMarkerIcon()
-        }
-
-        const marker = L.marker(latlng, {
+        let marker = L.marker(latlng, {
           icon: dynamicIcon(incident.properties?.icon),
           alt: incident.properties?.category.name,
           keyboard: false,
         })
+        // Matching on created_at since incidents do not have an ID
+        if (
+          selectedMarkerRef.current &&
```

Comment:
selectedMarkerRef?.current.feature?.properties.created_at ===
             incident.properties.created_at

if created_at is not nullable, and its actually required